features/api_full_auto_attach.feature

```diff
@@ -0,0 +1,54 @@
+Feature: Full Auto-Attach Endpoint
+
+    @series.lts
+    @uses.config.machine_type.aws.pro
+    @uses.config.machine_type.azure.pro
+    @uses.config.machine_type.gcp.pro
+    Scenario Outline: Run auto-attach on cloud instance.
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        contract_url: 'https://contracts.canonical.com'
+        data_dir: /var/lib/ubuntu-advantage
+        log_level: debug
+        log_file: /var/log/ubuntu-advantage.log
+        """
+        When I create the file `/tmp/full_auto_attach.py` with the following:
+        """
+        from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import full_auto_attach, FullAutoAttachOptions
+
+        full_auto_attach(FullAutoAttachOptions(enable=["esm-infra", "esm-apps"]))
+        """
+        And I run `python3 /tmp/full_auto_attach.py` with sudo
+        Then stdout matches regexp:
+        """
+        Ubuntu Pro: ESM Apps enabled
+        """
+        And stdout matches regexp:
+        """
+        Ubuntu Pro: ESM Infra enabled
+        """
```

Comment:
That script shouldn't print anything :upside_down_face: 

For now, this may have to be done by setting the event_logger to "json" mode, so that `event.info` calls don't print to stdout.

Ideally in the future, we'll break up all the individual pieces of auto attach so that none of them print in the actual action. Printing will only happen in the cli module. But that is the future state -  not for this PR

Reply:
CC @lucasmoura